test/config-luatest/cluster_config_test.lua

```diff
@@ -281,3 +281,33 @@ g.test_instance_uri_errors = function(g)
         end)
     end)
 end
+
+-- Attempt to pass an empty group and an empty replicaset.
+g.test_misplace_option = function(g)
+    local config = cbuilder.new()
+        :use_group('g-001')
+
+        :use_replicaset('r-001')
+        :add_instance('i-001', {})
+
+        :use_group('sharding')
+        :set_group_option('roles', {'super'})
```

Comment:
Nit: The original report contains the `sharding: {roles: [storage]}` field in a wrong place. Not `super`, but `storage` -- a sharding role.

An attempt to use `super` as a sharding role would fail the validation even if it is placed on a correct level. So, the test case has two errors.

I would use `storage` in the test to make it more clear what is the actual error we aim to verify.

Feel free to ignore, just thought.

Reply:
> I would use storage in the test to make it more clear what is the actual error we aim to verify.

Well, we expect a specific error message, so it's not a big deal, but for consistency I will update all examples, including the test.